docs/components/index.md

```diff
@@ -1,26 +1,156 @@
 <script setup>
-const components = import.meta.glob('./*/data.json', { eager: true })
-const baseUrl = import.meta.env.BASE_URL
+import { ref, computed } from 'vue';
 
-const componentData = Object.keys(components).map(path => {
- return ({
-  ...components[path].default, // component data from JSON
+// Import all JSON data
+const components = import.meta.glob('./*/data.json', { eager: true });
+const baseUrl = import.meta.env.BASE_URL;
+
+// Map JSON data to include href and correct paths
+const componentData = Object.keys(components).map(path => ({
+  ...components[path].default,
   href: `${baseUrl}components/${path.replace('/data.json', '')}`
-})});
+}));
+
+// Reactive variables for query and selected frameworks
+const query = ref('');
+const selectedFrameworks = ref([]);
```

Comment:
Would we like this list to be saved when we navigate between component pages and the overview page? Because right now the list gets reset whenever I refresh the overview page. If so, then maybe the ThemeSwitcher logic could help here, because we store the selected theme some place, too.

Reply:
Maybe...  could add that later, just as a note I think that can cause as many problems as it could help. People forgetting that the result is filtered/accidentally hitting the filter  and they dont see all anymore without realizing :D

Reply:
Then I'd let the users speak if they find it annoying :D I have a theory that no one using a desktop version of the docs will use this overview page anyway because they can already see an alphabetical list of components in the sidebar :D